### DISCUSSION TITLE
UR Bar, Progress meter and Skin Editor

### DISCUSSION THREAD

**Discussion by KevinMarkRoy** [ppy/osu#14234] [Ideas]
I had already made a [forum post](https://osu.ppy.sh/community/forums/topics/1393513) but a GMT suggested I make thread here too. I have a few suggestions which might affect gameplay visually.

1. Being able to Adjust the UR bar size, position, ect without having to use the skin editor or at least make it easier/more accessible. I understand that it's possible to do it technically but for someone who just starts using Lazer without knowing you have to use the Skin Editor to do this is an easy fix I think.
2. The ability to use the classic/stable version of the Progress meter from your skin instead of the lazer one. 
3. Accessing the Skin Editor is not so easy in that even if you know the shortcut is Ctrl+Shift+S, you have to do it while in game which is kind of odd since there are no indicators in the text boxes that popup if you try to use it in the menu that you have to use it in game. I get that all the UI stuff being modular in this way is still under development but I think the editor should be easier to use/more accessible to someone who just fired up the game coming from stable. 
4. Unable to edit the Legacy Combo element's position or size.

**Comment by bdach** [comment]
> Being able to Adjust the UR bar size, position, ect without having to use the skin editor

The skin editor should be the primary and *only* place for setting those properties.

> The ability to use the classic/stable version of the Progress meter from your skin instead of the lazer one.

Not implemented yet.

> you have to do it while in game which is kind of odd **since there are no indicators in the text boxes that popup if you try to use it in the menu that you have to use it in game**

I don't believe that is the case. See the big message in the middle of the screen as shown below.

![osu_2021-08-11_21-34-46](https://user-images.githubusercontent.com/20418176/129091923-f50af78f-6c7e-43c3-a380-7ecca327c2c7.jpg)

> Unable to edit the Legacy Combo element's position or size.

I don't follow? I can move the legacy combo counter just fine.

**Reply by KevinMarkRoy** [reply]
> The skin editor should be the primary and _only_ place for setting those properties.

This is not the case in Stable where you can change it in settings, but perhaps this is the way it should be done going forward.

> I don't believe that is the case. See the big message in the middle of the screen as shown below.
> ![osu_2021-08-11_21-34-46](https://user-images.githubusercontent.com/20418176/129091923-f50af78f-6c7e-43c3-a380-7ecca327c2c7.jpg)

"doesn't support skin customization yet" - reading this I believed that editing the skin was not possible at all at first. It does not say that you must do it while in gameplay to reach the skin editor for in-game elements.

> I don't follow? I can move the legacy combo counter just fine.

Might be an issue with my skin but when I was not able to move, resize or delete the combo counter in the skin editor.

**Reply by frenzibyte** [reply]
It sounds like your understanding of the skin editor is not correct, it wasn't made only to edit elements in gameplay, you'll eventually be able to edit certain elements displayed in every screen, but when a screen does not have any skinnable element displayed, you'll receive the "X doesn't support skin customization just yet."

**Reply by peppy** [reply]
Eventually we'll make it easier to navigate to a target screen for skinning from the editor. Probably with "sample" implementations of each screen, rather than first requiring navigation to the screen you want to edit. There's a lot of work left to be done to improve the UX of skin/layout editing.

**Reply by KevinMarkRoy** [reply]
> It sounds like your understanding of the skin editor is not correct, it wasn't made only to edit elements in gameplay, you'll eventually be able to edit certain elements displayed in every screen, but when a screen does not have any skinnable element displayed, you'll receive the "X doesn't support skin customization just yet."

Oh no I get that, its just that there are no indicators that you need to go in game to edit the gameplay elements. If you just press it when you're in song select or the front menu it just says "not supported yet".

**Reply by frenzibyte** [reply]
Re-read what I said, the skin editor tells you that editing X screen is not supported yet, with the name of the X screen being in a larger font, that should be enough to indicate that the skin editor edits on the active screen, and that you should go to gameplay for skin editor to edit on that.

Also as been mentioned above:
> Eventually we'll make it easier to navigate to a target screen for skinning from the editor. Probably with "sample" implementations of each screen, rather than first requiring navigation to the screen you want to edit. There's a lot of work left to be done to improve the UX of skin/layout editing.
